src/EditorFeatures/Core/Implementation/GoToDefinition/GoToDefinitionHelpers.cs

```diff
@@ -123,6 +145,55 @@ internal static class GoToDefinitionHelpers
             }
         }
 
+        public static async Task<IEnumerable<INavigableItem>> FindExternalDefinitionsAsync(Document document, int position, IEnumerable<Lazy<INavigableDefinitionProvider>> externalDefinitionProviders, CancellationToken cancellationToken)
+        {
+            foreach (var definitionProvider in externalDefinitionProviders)
+            {
+                var definitions = await definitionProvider.Value.FindDefinitionsAsync(document, position, cancellationToken).ConfigureAwait(false);
+                if (definitions != null && definitions.Any())
+                {
+                    var preferredDefinitions = NavigableItemFactory.GetPreferredNavigableItems(document.Project.Solution, definitions);
+                    if (preferredDefinitions.Any())
```

Comment:
We want to return the navigable definitions returned by first definition provider that has at least one preferred item, i.e. non-null Document.